Automattic/wp-calypso
# Domains: Add transfer lock to enhanced domain navigation

#### Changes proposed in this Pull Request

This PR adds the ability to determine the transfer lock status in the domain navigation menu.

Here's how the registration transfer lock looks like:

<img width="723" alt="Screenshot 2020-04-10 at 5 29 22 PM" src="https://user-images.githubusercontent.com/13062352/79004625-6aa73100-7b55-11ea-9ef5-e132af7a2b7c.png">

Transfer lock on:

<img width="720" alt="Screenshot 2020-04-10 at 5 29 40 PM" src="https://user-images.githubusercontent.com/13062352/79004640-74c92f80-7b55-11ea-80aa-ee7ea162a1ad.png">

Transfer lock off:

<img width="718" alt="Screenshot 2020-04-10 at 5 30 44 PM" src="https://user-images.githubusercontent.com/13062352/79004642-772b8980-7b55-11ea-8ab5-abcceaf90a95.png">


Depends on D41708-code

#### Testing instructions

- Apply the backend patch
- If you have a domain that's been registered within the last month or so, check that the correct message appears with the right date (similar to the first screenshot above)
- For a domain registered for more than 60 days, check that the transfer lock appears correctly
